tests/platform_tests/mellanox/check_sysfs.py

```diff
@@ -227,3 +227,26 @@ def check_psu_sysfs(dut, psu_id, psu_state):
         assert (psu_pwr_state_content["stdout"] == "1" and psu_state == "OK") \
                 or (psu_pwr_state_content["stdout"] == "0" and psu_state == "NOT OK"),\
             "sysfs content %s mismatches with psu_state %s" % (psu_pwr_state_content["stdout"], psu_state)
+
+
+def _check_fan_speed_in_tolerance(dut, fan_speed, min_speed, max_speed, sysfs_path):
+    read_count = 6
+    read_interval = 5
+    in_range = False
+    while 1:
+        if min_speed < fan_speed < max_speed:
+            in_range = True
+            break
+
+        read_count -= 1
+        if read_count <= 0:
+            break
+
+        time.sleep(read_interval)
+        try:
+            fan_speed_get_content = dut.command("cat %s" % sysfs_path)
+            fan_speed = int(fan_speed_get_content["stdout"])
+        except Exception as e:
+            assert "Get content from %s failed, exception: %s" % (sysfs_path, repr(e))
```

Comment:
Can you take a look at the `wait_until` method here and see if you can reuse it for this purpose?
https://github.com/Azure/sonic-mgmt/blob/f73099c32a25bdbc21981123fa3fd49ebeea09bd/tests/common/utilities.py#L18